sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/DefaultHttpClient.java

```diff
@@ -209,26 +216,148 @@ private HttpResponse receiveResponse(HttpRequest httpRequest, HttpURLConnection
-                BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+                return new DefaultHttpClientResponse(httpRequest, responseCode, responseHeaders,
+                    BinaryData.fromByteBuffer(outputStream.toByteBuffer()));
+            }
         } catch (IOException e) {
             throw LOGGER.logThrowableAsError(new RuntimeException(e));
+
         } finally {
             connection.disconnect();
         }
     }
 
+    /**
+     * Processes the sse buffer and dispatches the event
+     * @param reader The BufferedReader object
+     * @throws IOException If an I/O error occurs
+     */
+    private void processBuffer(BufferedReader reader) throws IOException {
+        StringBuilder sb = new StringBuilder(), data = new StringBuilder();
+        String eventStr = null;
+        long id = -1;
+        ServerSentEvent event = new ServerSentEvent();
+        int dataRead = reader.read();
+        while (dataRead != -1) {
+            sb.append((char) dataRead);
+            dataRead = reader.read();
+            int index;
+            while ((index = sb.indexOf("\n\n")) >= 0) { // blocks of data are separated by double newlines
+                String[] lines = sb.substring(0, index).split("\n");
+                sb.delete(0, index + 2); // clear first block including "\n\n"
+                boolean hasDataOrEvent = false;
+                for (String line : lines) { //start parsing line by line
+                    int idx = line.indexOf(':');
+                    if (idx < 0) {
+                        continue; // ignore invalid data
+                    } else if (idx == 0) { // capture comment line
+                        event = event.setComment(line.substring(1).trim());
+                    }
+                    String field = line.substring(0, idx), value = line.substring(idx + 1).trim();
+                    switch (field.trim().toLowerCase()) {
+                        case "event":
+                            eventStr = value;
+                            hasDataOrEvent = true;
+                            break;
+
+                        case "data":
+                            if (data.length() > 0) data.append("\n");
+                            data.append(value);
+                            hasDataOrEvent = true;
+                            break;
+
+                        case "id":
+                            id = Long.parseLong(value);
+                            lastEventID = id;
+                            event = event.setId(id);
+                            break;
+
+                        case "retry":
+                            if (DIGITS_ONLY.matcher(value).matches()) {
+                                retryAfter = Long.parseLong(value);
+                            }
+                            event = event.setRetryAfter(retryAfter);
+                            break;
+
+                        default:
+                            break;
```

Comment:
We should log this and throw an error if we get an unexpected field.

Reply:
Right now in azure-core we don't have azure-json or Jackson log if there is an unknown property. Should those log as well or should we leave this unlogged since there isn't much a user could do with that information? Or, possibly more convenient, have an unknown properties bag in the model